android-components/components/browser/state/src/main/java/mozilla/components/browser/state/reducer/TranslationsStateReducer.kt

```diff
@@ -77,14 +77,26 @@ internal object TranslationsStateReducer {
                 }
 
                 TranslationOperation.FETCH_LANGUAGES -> {
-                    // Generally expect [TranslationsAction.TranslateSetLanguagesAction] to be used
-                    // as the success condition since the updated value is needed.
+                    // Reset the error state, and then generally expect
+                    // [TranslationsAction.TranslateSetLanguagesAction] to update state in the
+                    // success case.
                     state.copyWithTranslationsState(action.tabId) {
                         it.copy(
                             translationError = null,
                         )
                     }
                 }
+
+                TranslationOperation.FETCH_PAGE_SETTINGS -> {
```

Comment:
Aside: I almost feel like we need a diagram of the action flows.

Reply:
+1, I'll prioritize this documentation. It'll help with planning the remaining actions and coordinating with other engineers.